include/StringTools.h

```diff
@@ -0,0 +1,294 @@
+    /// A "*" (asterisk) will match 0 or more characters of any value. @n
+    /// A "?" (question mark) will match 0 or 1 character of any value.
+    /// @param Source The string to check.
+    /// @param Pattern The pattern to try and match to the source string.
+    /// @param CaseSensitive Whether or not the pattern should match case exactly.
+    /// @return Returns true if the source string matches the specified pattern.
+    Boole MEZZ_LIB PatternMatch(const StringView Source, const StringView Pattern, const Boole CaseSensitive);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Convert-From-String Functions
+
+    /// @brief Converts a String to any.
+    /// @tparam Return The type the String will be converted to.
+    /// @remarks This function should work for any type with an input streaming operator.
+    /// @param ToConvert The String that will be converted.
+    /// @return Returns the value from the converted String.
+    template<typename Return>
+    Return ConvertFromString(const StringView ToConvert)
+    {
+        Return Converted{};
+        StringStream Converter;
+        Converter << ToConvert;
+        Converter >> Converted;
+        return Converted;
+    }
+
+    /// @brief Converts a String into a Boole.
+    /// @param ToConvert The String to be converted to a Boole.
+    /// @return Returns a Boole with the converted value.
+    Boole MEZZ_LIB ConvertToBool(const StringView ToConvert, const Boole Default = false);
+    /// @brief Converts a String to an Int8.
+    /// @remarks This requires its own specialization because otherwise a Int8 will use the char overload.
+    /// @param ToConvert The String to be converted to an Int8.
+    /// @return Returns an Int8 with the converted value.
+    template<>
+    Int8 ConvertFromString<Int8>(const StringView ToConvert);
+    /// @brief Converts a String to a UInt8.
+    /// @remarks This requires its own specialization because otherwise a UInt8 will implicitly covert to a char
+    /// and use that overload.
+    /// @param ToConvert The String to be converted to a UInt8.
+    /// @return Returns a UInt8 with the converted value.
+    template<>
+    UInt8 ConvertFromString<UInt8>(const StringView ToConvert);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Convert-To-String Functions
+
+    /// @brief Converts any into a String.
+    /// @tparam T The deduced type that will be converted to a String.
+    /// @remarks This function should work for any type with an output streaming operator.
+    /// @param ToConvert Streamable type instance to be converted.
+    /// @return Returns a String containing the lexicographically converted data.
+    template<typename T>
+    String ConvertToString(const T& ToConvert)
+    {
+        StringStream Converter;
+        Converter << ToConvert;
+        return Converter.str();
+    }
```

Comment:
Should This delegate to std::to_string?

Reply:
No.  std::to_string is intended for numeric conversions.  Whereas this will work with any type that has a streaming operator.